Example/GiniVision/ComponentAPICoordinator.swift

```diff
@@ -168,151 +206,323 @@ final class ComponentAPICoordinator: NSObject, Coordinator {
+        // Here you can show the Onboarding screen in case that you decide
+        // to launch it once the camera screen appears
+    }
+    
+    func cameraDidTapMultipageReviewButton(_ viewController: CameraViewController) {
+        showMultipageReviewScreen()
+    }
+    
+    func camera(_ viewController: CameraViewController, didSelect documentPicker: DocumentPickerType) {
+        switch documentPicker {
+        case .gallery:
+            documentPickerCoordinator.showGalleryPicker(from: viewController)
+        case .explorer:
+            documentPickerCoordinator.isPDFSelectionAllowed = documentRequests.isEmpty
+            documentPickerCoordinator.showDocumentPicker(from: viewController)
+        case .dragndrop: break
+        }
     }
 }
 
-// MARK: ComponentAPIReviewScreenDelegate
+// MARK: - DocumentPickerCoordinatorDelegate
 
-extension ComponentAPICoordinator: ComponentAPIReviewViewControllerDelegate {
-    func componentAPIReview(viewController: ComponentAPIReviewViewController,
-                            didReviewDocument document: GiniVisionDocument) {
-        // Present already existing results retrieved from the first analysis process initiated in `viewDidLoad`.
-        if let result = documentService.result,
-            let document = documentService.document {
-            handleAnalysis(result, fromDocument: document)
-            return
-        }
-        
-        // Restart analysis if it was canceled and is currently not running.
-        if !documentService.isAnalyzing {
-            if let documentData = self.document?.data {
-                documentService.analyzeDocument(withData: documentData,
-                                                cancelationToken: CancelationToken(),
-                                                completion: didDocumentAnalysisComplete)
+extension ComponentAPICoordinator: DocumentPickerCoordinatorDelegate {
+    func documentPicker(_ coordinator: DocumentPickerCoordinator, didPick documents: [GiniVisionDocument]) {
+        self.validate(documents) { result in
+            switch result {
+            case .success(let documentRequests):
+                coordinator.dismissCurrentPicker {
+                    self.documentRequests.append(contentsOf: documentRequests)
+                    self.upload(documentRequests: documentRequests)
+                    self.showNextScreenAfterPicking()
+                }
+            case .failure(let error):
+                var positiveAction: (() -> Void)?
+                
+                if let error = error as? FilePickerError {
+                    switch error {
+                    case .maxFilesPickedCountExceeded, .mixedDocumentsUnsupported:
+                        if !self.documentRequests.isEmpty {
+                            positiveAction = {
+                                coordinator.dismissCurrentPicker {
+                                    self.showMultipageReviewScreen()
```

Comment:
And this two-humped camel of doom is just ridiculous!

Reply:
Again I cannot agree with you, this is perfectly readable and extracting to a different method would make it worse.